devbin/generate_gcp_ar_cleanup_policy.py

```diff
@@ -0,0 +1,140 @@
+                 newer_than: Optional[str] = None):
+        self.name = name
+        self.tag_state = tag_state
+        self.tag_prefixes = tag_prefixes
+        self.version_name_prefixes = version_name_prefixes
+        self.package_name_prefixes = package_name_prefixes
+        self.older_than = older_than
+        self.newer_than = newer_than
+
+    def to_dict(self):
+        data = {'name': self.name, 'action': {'type': 'Keep'}, 'condition': {'tagState': self.tag_state}}
+        condition = data['condition']
+        if self.tag_prefixes is not None:
+            condition['tagPrefixes'] = self.tag_prefixes
+        if self.version_name_prefixes is not None:
+            condition['versionNamePrefixes'] = self.version_name_prefixes
+        if self.package_name_prefixes is not None:
+            condition['packageNamePrefixes'] = self.package_name_prefixes
+        if self.older_than:
+            condition['olderThan'] = self.older_than
+        if self.newer_than:
+            condition['newerThan'] = self.newer_than
+        return data
+
+
+class MostRecentVersionKeepPolicy(CleanupPolicy):
+    def __init__(self,
+                 name: str,
+                 package_name_prefixes: List[str],
+                 keep_count: int):
+        self.name = name
+        self.package_name_prefixes = package_name_prefixes
+        self.keep_count = keep_count
+
+    def to_dict(self):
+        data = {
+            'name': self.name,
+            'action': {'type': 'Keep'},
+            'mostRecentVersions': {
+                'packageNamePrefixes': self.package_name_prefixes,
+                'keepCount': self.keep_count
+            }
+        }
+        return data
+
+
+third_party_images_fp = 'docker/third-party/images.txt'
+third_party_packages = []
+with open(third_party_images_fp, 'r') as f:
+    for image in f:
+        image = image.strip()
+        package = image.split(':')[0]
+        if package not in third_party_packages:
+            third_party_packages.append(package)
+
+
+deploy_packages = []
+with open('build.yaml', 'r') as f:
+    config_str = f.read().strip()
+    build_config = yaml.safe_load(config_str)
+    for step in build_config['steps']:
+        if step['kind'] == 'buildImage2':
+            image = step['publishAs']
+            if image not in deploy_packages:
+                deploy_packages.append(image)
+
+
+third_party_packages.sort()
+deploy_packages.sort()
+
+policies = [
+    DeletePolicy('delete_untagged', 'untagged'),
+    DeletePolicy('delete_dev', 'tagged', tag_prefixes=['dev-'], older_than='3d'),
+    DeletePolicy('delete_test_pr', 'tagged', tag_prefixes=['test-pr-'], older_than='3d'),
+    DeletePolicy('delete_test_deploy', 'tagged', tag_prefixes=['test-deploy-'], older_than='3d'),
+    DeletePolicy('delete_cache', 'tagged', tag_prefixes=['cache-pr-'], older_than='7d'),
+    DeletePolicy('delete_cache', 'tagged', tag_prefixes=['cache-'], older_than='30d'),
```

Comment:
Are the names required to be unique?

Reply:
I sort above.

Reply:
I'm referring to the names of the policies themselves; `delete_cache` appears twice.